test/gtest/ucp/test_ucp_mem_type.cc

```diff
@@ -87,6 +87,9 @@ class test_ucp_mem_type_alloc_before_init : public test_ucp_mem_type {
     virtual void init() {
         m_send_buffer.reset(new mem_buffer(m_size, mem_type()));
         m_recv_buffer.reset(new mem_buffer(m_size, mem_type()));
+
+        m_send_buffer->set(m_send_buffer->ptr(), 0, m_size, mem_type());
```

Comment:
maybe use pattern_fill + pattern_check ?

Reply:
yep, this is good idea

Reply:
> maybe use pattern_fill + pattern_check ?

@yosefe done, CUDA support is ok there